geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/PulseAbstractTest.java

```diff
@@ -136,6 +146,29 @@
 
   private static final DecimalFormat df2 = new DecimalFormat(PulseConstants.DECIMAL_FORMAT_PATTERN);
 
+  @Rule
+  public TestRule testWatcher = new TestWatcher() {
+    @Override
+    public void failed(Throwable t, Description test) {
+      takeScreenshot(test.getDisplayName());
+    }
+  };
+
+  @Rule
+  public RetryRule retryRule = new RetryRule(5);
+
+  public void takeScreenshot(String screenshotName) {

```

Comment:
I found screenshots to be helpful when I was debugging the tests that failed from switching to phantomjs.  Maybe we could keep this code around but just keep the screenshot testWatcher off by default so that people can turn them on if desired?